system-tests/test.py

```diff
@@ -145,12 +186,21 @@ def _assert_diff(self, first, second):
     #
     root_dir = Path(__file__).parent
 
-    sequence = [
-        FabTestCase(root_dir / 'MinimalFortran'),
-        FabTestCase(root_dir / 'FortranDependencies'),
-        FabTestCase(root_dir / 'FortranPreProcess',
-                    fpp_flags='-DSHOULD_I_STAY=yes')
+    sequence = (
+        [
+            FabTestCase(root_dir / 'MinimalFortran'),
+            DumpTestCase(root_dir / 'MinimalFortran')
+        ],
+        [
+            FabTestCase(root_dir / 'FortranDependencies'),
+            DumpTestCase(root_dir / 'FortranDependencies')
+        ],
+        [
+            FabTestCase(root_dir / 'FortranPreProcess',
+                        fpp_flags='-DSHOULD_I_STAY=yes'),
+            DumpTestCase(root_dir / 'FortranPreProcess')
```

Comment:
So just checking - these are executed in order?  So the `DumpTestCase` will be running on the pre-processed result?  (that seems to be the case based on the expected output files)

Reply:
That is correct. I have commented the structure to make it clearer.